mission_control/navigator_missions/nav_missions/3Ddetect.py

```diff
@@ -0,0 +1,210 @@
+    cv2.line(img, tuple(corner), tuple(imgpts[1].ravel()), (0,255,0), 5)
+    cv2.line(img, tuple(corner), tuple(imgpts[2].ravel()), (0,0,255), 5)
+    return img
+
+#~ def getBoundingTrapCross(points):
+    #~ for i in range(len(points)):
+def getCenter(points):
+    cx = 0
+    cy = 0
+    for i in range(len(points)):
+        cx += points[i][0]
+        cy += points[i][1]
+    cx = cx/len(points)
+    cy = cy/len(points)
+    return np.array([cx,cy])
+        
+
+def getMax(points, what):
+    max_val = 0
+    second_max = 0
+    if(points[0][what] > points[1][what]):
+        second_max = points[1];
+        max_val = points[0];
+    else:
+        second_max = points[0];
+        max_val = points[1];
+    
+    for i in range(2,len(points)):
+        if(points[i][what] >= max_val[what]):
+            second_max=max_val;
+            max_val=points[i];          
+        elif(points[i][what] > second_max[what]):
+            second_max=points[i];
+    
+    return max_val[:-1], second_max[:-1]
+
+def getMin(points, what):
+    min_val = 2000
+    second_min = 2000
+    if(points[0][what] < points[1][what]):
+        second_min = points[1];
+        min_val = points[0];
+    else:
+        second_min = points[0];
+        min_val = points[1];
+    
+    for i in range(2,len(points)):
+        if(points[i][what] <= min_val[what]):
+            second_min=min_val;
+            min_val=points[i];          
+        elif(points[i][what] < second_min[what]):
+            second_min=points[i];
+    
+    return min_val[:-1], second_min[:-1]
+
+def perp(a):
+    b = np.empty_like(a)
+    b[0] = -a[1]
+    b[1] = a[0]
+    return b
+
+def seg_intersect(a1,a2, b1,b2):
+    da = a2-a1
+    db = b2-b1
+    dp = a1-b1
+    dap = perp(da)
+    denom = np.dot( dap, db)
+    num = np.dot( dap, dp )
+    return (num / denom.astype(float))*db + b1
+
+def getPlanarBoundingRect(points):
+    bigY = getMax(points, 1)
+    bigX = getMax(points, 0)
+    
+    smallY = getMin(points,1)
+    smallX = getMin(points,0)
+    
+    point1=tuple(seg_intersect(bigX[0], bigX[1], bigY[0], bigY[1]).astype(int))
+    point2=tuple(seg_intersect(smallX[0], smallX[1], smallY[0], smallY[1]).astype(int))
+    point3=tuple(seg_intersect(smallX[0], smallX[1], bigY[0], bigY[1]).astype(int))
+    point4=tuple(seg_intersect(bigX[0], bigX[1], smallY[0], smallY[1]).astype(int))
+
+    return point1, point2, point3, point4
+    
+
+def boundingRect(points):
+    maxX = 0
+    minX = 2000
+    maxY = 0
+    minY = 2000
+    for i in range(len(points)):
+        if(points[i][0] > maxX):
+            maxX = points[i][0];
+        if(points[i][1] > maxY):
+            maxY = points[i][1];
+        if(points[i][0] < minX):
+            minX = points[i][0];
+        if(points[i][1] < minY):
+            minY = points[i][1];
+    return np.array([[maxX, maxY], [minX, minY],[maxX,minY],[minX,maxY]])
+
+@txros.util.cancellableInlineCallbacks
+def main(navigator):
```

Comment:
This looks like it's entirely test code - it shouldn't be in the master repo IMO